sdk/eventhub/azure-eventhubs/api_review.py

```diff
@@ -0,0 +1,235 @@
+
+    async def get_properties(self) -> Dict[str, Any]:
+
+    async def get_partition_ids(self) -> List[str]:
+
+    async def get_partition_properties(self, partition: str) -> Dict[str, str]:
+
+    async def close(self) -> None:
+
+
+class PartitionContext(object):
+
+    fully_qualified_namespace  # type: str
+    partition_id  # type: str
+    eventhub_name  # type: str
+    consumer_group_name  # type: str
+    owner_id  # type: str
+
+    def __init__(
+        self,
+        fully_qualified_namespace: str,
+        eventhub_name: str,
+        consumer_group_name: str,
+        partition_id: str,
+        owner_id: str,
+        partition_manager: PartitionManager =None) -> None:
+
+    async def update_checkpoint(self, event: EventData):
+
+
+class PartitionManager(ABC):
+
+    async def list_ownership(
+        self,
+        fully_qualified_namespace: str,
+        eventhub_name: str,
+        consumer_group_name: str) -> Iterable[Dict[str, Any]]:
+
+    async def claim_ownership(self, ownership_list: Iterable[Dict[str, Any]]) -> Iterable[Dict[str, Any]]:
+
+    async def update_checkpoint(
+        self,
+        fully_qualified_namespace: str,
+        eventhub_name: str,
+        consumer_group_name: str,
+        partition_id: str,
+        offset: str,
+        sequence_number: int) -> None:
+
+    async def list_checkpoints(self, fully_qualified_namespace: str, eventhub_name: str, consumer_group_name: str) -> Iterable[Dict[str,Any]]:
+
+
+class EventData(object):
+
+    message  # type: uamqp.Message
+
+    def __init__(self, body: Union[str, bytes] =None) -> None:
+
+    def __str__(self) -> str:
+
+    @property
+    def sequence_number(self) -> int:
+
+    @property
+    def offset(self) -> str:
+
+    @property
+    def enqueued_time(self) -> datetime.datetime:
+    
+    @property
+    def partition_key(self) -> bytes:
+
+    @property
+    def application_properties(self) -> dict[str, Any]:
+
+    @property
+    def system_properties(self) -> dict[str, Any]:
+
+    @property
+    def body(self) -> Union[bytes, Iterable[bytes]]:
+
+    @property
+    def last_enqueued_event_properties(self) -> dict[str, Any]:
+
+    def body_as_str(self, encoding='UTF-8') -> str:
+
+    def body_as_json(self, encoding='UTF-8') -> dict[str, Any]:
+
+    def encode_message(self) -> bytes:
+
+
+class EventDataBatch(object):
+
+    message  # type: uamqp.BatchMessage
+    max_size  # type: int
+
+    def __init__(self, max_size: int=None, partition_key: str=None) -> None:
+    
+    def __len__(self) -> int:
+
+    @property
+    def size(self) -> int:
+
+    def try_add(self, event_data: EventData) -> None:
+
+
+class EventPosition(object):
+
+    value  # type: Union[str, int, datetime.datetime]
+    inclusive  # type: bool
+
+    def __init__(self, value, inclusive=False):
+
+
+class CloseReason(Enum):
+    SHUTDOWN = 0
+    OWNERSHIP_LOST = 1
+
+
+class EventHubSharedKeyCredential(object):
+
+    def __init__(self, policy: str, key: str):
```

Comment:
There's some variation here for naming: policy, policy_name, key_name
Pending a side-by-side comparison.